tests/foreman/cli/test_auth.py

```diff
@@ -0,0 +1,415 @@
+        Auth.login({'username': self.uname_admin, 'password': self.password})
+        result = Auth.with_user().status()
+        self.assertIn(
+            LOGEDIN_MSG.format(self.uname_admin),
+            result[0][u'message']
+        )
+        # list organizations without supplying credentials
+        with self.assertNotRaises(CLIReturnCodeError):
+            Org.with_user().list()
+        Auth.logout()
+        result = Auth.with_user().status()
+        self.assertIn(
+            LOGEDOFF_MSG.format(self.uname_admin),
+            result[0][u'message']
+        )
+        with self.assertRaises(CLIReturnCodeError):
+            Org.with_user().list()
+
+    @tier1
+    def test_positive_change_session(self):
+        """Change from existing session to a different session
+
+        :id: b6ea6f3c-fcbd-4e7b-97bd-f3e0e6b9da8f
+
+        :Steps:
+
+            1. Set use_sessions
+            2. Authenticate, assert credentials are not demanded
+               on next command run
+            3. Login as a different user
+
+        :expectedresults: The session is altered
+
+        """
+        result = configure_sessions()
+        self.assertEqual(result, 0, 'Failed to configure hammer sessions')
+        Auth.login({'username': self.uname_admin, 'password': self.password})
+        result = Auth.with_user().status()
+        self.assertIn(
+            LOGEDIN_MSG.format(self.uname_admin),
+            result[0][u'message']
+        )
+        # list organizations without supplying credentials
+        with self.assertNotRaises(CLIReturnCodeError):
+            Org.with_user().list()
+        Auth.login({'username': self.uname_viewer, 'password': self.password})
+        result = Auth.with_user().status()
+        self.assertIn(
+            LOGEDIN_MSG.format(self.uname_viewer),
+            result[0][u'message']
+        )
+        with self.assertNotRaises(CLIReturnCodeError):
+            Org.with_user().list()
+
+    @tier1
+    def test_positive_session_survives_unauthenticated_call(self):
+        """Check if session stays up after unauthenticated call
+
+        :id: 8bc304a0-70ea-489c-9c3f-ea8343c5284c
+
+        :Steps:
+
+            1. Set use_sessions
+            2. Authenticate, assert credentials are not demanded
+               on next command run
+            3. Run `hammer ping`
+
+        :expectedresults: The session is unchanged
+
+        """
+        result = configure_sessions()
+        self.assertEqual(result, 0, 'Failed to configure hammer sessions')
+        Auth.login({'username': self.uname_admin, 'password': self.password})
+        result = Auth.with_user().status()
+        self.assertIn(
+            LOGEDIN_MSG.format(self.uname_admin),
+            result[0][u'message']
+        )
+        # list organizations without supplying credentials
+        with self.assertNotRaises(CLIReturnCodeError):
+            Org.with_user().list()
+        ssh.command('hammer ping')
```

Comment:
I'm not sure i've got the idea of this test right, but `ssh.command` doesn't perform any checks of stdout/stderr/return_code, this `hammer ping` command could fail and without assertion we'll never catch that. Maybe assertions are needed here?

Reply:
think we should assert on return code